pkg/controller/jobframework/reconciler.go

```diff
@@ -333,8 +347,16 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object clie
 		}
 	}
 
-	if wl != nil {
-		job.RestoreNodeAffinity(wl.Spec.PodSets)
+	log.V(3).Info("restore node selectors from annotation")
+	selectors, err := getNodeSelectorsFromObjectAnnotation(object)
+	if err != nil {
+		if errors.Is(err, errNodeSelectorsNotFound) {
+			log.V(3).Info("No original node selectors found")
```

Comment:
Why do we distiguish the two cases? It seems the absence of the annotation is an indication of an error? Is there any scenario under which the annotation might be not set?

Reply:
Yes, at the initial `stopJob`, before any workload is created, or if the workload is deleted before being accepted.

Reply:
I don't think this is resolved. Please provide more details for the scenarios when the annotation is missing when `stopJob` is called (sequence of steps). 

If there is such a scenario, please add a comment in code so that it is clear for future readers. 

If we don't have a scenario then there is no need for a special case for the Info log - use error always. Also, the we can clean up the comment for `OriginalNodeSelectorsAnnotation`.

Reply:
The one case in which the missing annotation is acceptable is when the job is created unsuspended (in my opinion is guaranteed the the web-hook will always work for any kind of job, also the MPIJob integration tests do mot register webhooks) . However let's make them all errors.